tensorflow_probability/python/mcmc/diagnostic.py

```diff
@@ -492,6 +493,18 @@ def potential_scale_reduction(chains_states,
 def _potential_scale_reduction_single_state(state, independent_chain_ndims,
                                             split_chains, validate_args):
   """potential_scale_reduction for one single state `Tensor`."""
+  # casting integers to floats for floating-point division
+  # check to see if the `state` is a numpy object for the numpy test suite
+  if type(state).__module__ == np.__name__:
```

Comment:
This seems very kluge-like to me of a fix. Would be nice if there was a general check for both numpy and tf ints

Reply:
We have dtype_util.is_integer